src/ipa_pc/mod.rs

```diff
@@ -736,6 +736,136 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         })
     }
 
+    /// Refers to the "Multi-poly multi-point assertions" section of the "Recursive SNARKs based on Marlin" document
+    fn batch_open_individual_opening_challenges<'a>(
+        ck: &Self::CommitterKey,
+        labeled_polynomials: impl Clone + IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
```

Comment:
I don't like very much the addition of Clone to the impl in several places, but I experimented with it myself a little bit and saw that lifetimes makes difficult to do otherwise. If you can't find another solution just keep it as it is

Reply:
I've removed all this "Clone +" from functions signatures. Also I've rewrite the code so the iter() instead of into_iter() is used, so we don't need to clone the data structures several times during the computation